lib/Transforms/SetSPIRVCapabilities.cpp

```diff
@@ -53,8 +55,10 @@ struct SetSPIRVCapabilitiesPass
         triple, spirv::Vendor::Unknown, spirv::DeviceType::Unknown,
         spirv::TargetEnvAttr::kUnknownDeviceID,
         spirv::getDefaultResourceLimits(context));
-    auto module = getOperation();
-    module->setAttr(spirv::getTargetEnvAttrName(), attr);
+    auto op = getOperation();
+    op->walk([&](mlir::gpu::GPUModuleOp op) {
```

Comment:
this would walk through all gpu modules and sets spv attribute, is this the only attribute that we need to take care? how did you find this out?

Reply:
it was complaining about it while running e2e through runner